driverapi/driverapi.go

```diff
@@ -72,6 +72,11 @@ type Driver interface {
 	// only invoked for the global scope driver.
 	EventNotify(event EventType, nid string, tableName string, key string, value []byte)
 
+	// DecodeTableEntry passes the driver a key, value pair from table it registered
+	// with libnetwork and expects the driver to return a {endpoint ID, info string}
```

Comment:
@mavenugo `But the DecodeTableEntry seems to have it with an assumption that overlay driver depends on endpoint-id.` This is not correct. The reason why DecodeTableEntry expects a tuple {endpoint ID, info} back is precisely because libnetwork can't make that assumption. In future if a driver starts using networkDB for some random data unrelated to endpoint when `DecodeTableEntry` is called from the table name and key it can identify what is the data being stored and return an empty string, "",""

`to narrow down the AddTableEntry and EventNotify to carry the endpoint-id as well` - I don't think that would be a good API. What do we do for drivers that want to use networkDB for non-endpoint related data ? We don't want to define multiple flavors of AddTableEntry/EventNotify. 

I think what we can do is add an option to `TableEventRegister` to indicate what object the drivers want to store in that table, the options can be ENDPOINT_OBJECT, NETWORK_OBJECT, OPAQUE. So in this scenario for example, libnetwork will call DecodeTableEntry only for the table that is associated with ENDPOINT_OBJECT.

Since we haven't exposed networkdb yet to drivers may be this is a change we can make now without any backward compatibility issues. But please note that as explained above this PR doesn't make any hard assumptions and the network API enhancement can be done independently.